### PR TITLE
complete files.stat with the 'with-local' option

### PR DESCRIPTION
**Pull request by MichaelMure**
Following https://github.com/ipfs/go-ipfs/pull/4638, this new option is present in go-ipfs that enable to know if and how much of a dag is present locally.

Now, in the go-ipfs's http API, `WithLocality`, `Local` and `SizeLocal` in the answer are not present when `with-local` is not active. I chose to set default values in the javascript object response, but it could also be done the same way with no values present at all. Which would be best ?

### CODE REVIEW

File: SPEC/FILES.md [hunk 1]
```diff
@@ -618,6 +618,7 @@ Where:
 - `options` is an optional Object that might contain the following keys:
   - `hash` is a Boolean value to return only the hash.
   - `size` is a Boolean value to return only the size.
+  - `with-local` is a Boolean value to compute the amount of the dag that is local, and if possible the total size.
```
Comment (diasdavid) [inline SPEC/FILES.md]: When options have `-`, we translate them to camel case. `withLocal`

Comment (MichaelMure) [reply SPEC/FILES.md]: Will do, but it's currently not respected in js-ipfs-api. For instance: https://github.com/ipfs/js-ipfs-api/blob/master/test/files.spec.js#L93

Comment (diasdavid) [reply SPEC/FILES.md]: Seems that one passed by review. That said, we still want to keep the camelCase version and support the dash for backwards compatibility

Comment (MichaelMure) [reply SPEC/FILES.md]: I see that lodash is used exactly once in js-ipfs-api. Nothing against using `opts = _.mapKeys(opts, (v, k) => _.kebabCase(k))` ?

That would support both version and some more easily.

File: js/src/files-mfs.js [hunk 2]
```diff
@@ -295,7 +298,54 @@ module.exports = (common) => {
...
+            sizeLocal: undefined
+          })
+          done()
+        })
+      })
+
+      it('stat withLocal file', (done) => {
+        if (!withGo) {
+          console.log('Not supported in js-ipfs yet')
+          return done()
```
Comment (diasdavid) [inline js/src/files-mfs.js]: call `this.skip()`. You will have to replace the arrow function by regular function

File: js/src/files-mfs.js [hunk 3]
```diff
@@ -295,7 +298,54 @@ module.exports = (common) => {
...
+          })
+          done()
+        })
+      })
+
+      it('stat withLocal dir', (done) => {
+        if (!withGo) {
+          console.log('Not supported in js-ipfs yet')
+          return done()
+        }
```
Comment (diasdavid) [inline js/src/files-mfs.js]: same as above.

File: js/src/files.js [hunk 4]
```diff
@@ -980,5 +985,31 @@ module.exports = (common) => {
...
       })
     })
+
+    describe('.stat', () => {
+      before((done) => ipfs.files.add(smallFile.data, done))
+
+      it('stat outside of mfs', () => {
+        if (!withGo) {
+          console.log('Not supported in js-ipfs yet')
+          return done()
```
Comment (diasdavid) [inline js/src/files.js]: same as above.

File: js/src/files-mfs.js (hunk 2 above)
Comment (MichaelMure) [reply js/src/files-mfs.js]: This is the way it's done in all the codebase. Do you prefer to change that here or in a single pass in another PR ?

Comment (diasdavid) [reply js/src/files-mfs.js]: can you change the others first in a PR and update this one to match? thank you :)

**Comment by MichaelMure** [comment]
@diasdavid I changed for `this.skip()` as well so it should be fine now

**Comment by vmx** [comment]
This PR broke most of the `.stat` tests run by js-ipfs-api: https://ci.ipfs.team/blue/organizations/jenkins/IPFS%2Fjs-ipfs-api/detail/get-ci-green/1/tests/

@MichaelMure Can you please have a look?

**Comment by diasdavid** [comment]
@vmx you will need https://github.com/ipfs/js-ipfs-api/pull/695. Wanna update deps there? 

This is one of the reasons why I asked you for a issue that lists all the PR to make js-ipfs-api tests. There are so many PR's in flight that is hard to manage.

**Comment by MichaelMure** [comment]
@vmx Thank for taking care of that.

**Comment by vmx** [comment]
@MichaelMure  It wasn't me, @diasdavid fixed those tests.

**Comment by MichaelMure** [comment]
Errr, thanks @diasdavid ^^

**Pull request merged**